temporalio/worker.py

```diff
@@ -0,0 +1,1107 @@
+                started_time=_proto_maybe_datetime(start.started_time)
+                if start.HasField("started_time")
+                else None,
+                task_queue=self._config["task_queue"],
+                task_token=task_token,
+                workflow_id=start.workflow_execution.workflow_id,
+                workflow_namespace=start.workflow_namespace,
+                workflow_run_id=start.workflow_execution.run_id,
+                workflow_type=start.workflow_type,
+            )
+            input = ExecuteActivityInput(
+                fn=activity_def.fn,
+                args=args,
+                executor=None
+                if inspect.iscoroutinefunction(activity_def.fn)
+                else self._config["activity_executor"],
+                _cancelled_event=running_activity.cancelled_event,
+                _worker_shutdown_event=running_activity.worker_shutdown_event,
+                _worker=self,
+            )
+
+            # Set the context early so the logging adapter works and
+            # interceptors have it
+            temporalio.activity._Context.set(
+                temporalio.activity._Context(
+                    info=lambda: info,
+                    heartbeat=None,
+                    cancelled_event=running_activity.cancelled_event,
+                    worker_shutdown_event=running_activity.worker_shutdown_event,
+                )
+            )
+            temporalio.activity.logger.debug("Starting activity")
+
+            # Build the interceptors chaining in reverse. We build a context right
+            # now even though the info() can't be intercepted and heartbeat() will
+            # fail. The interceptors may want to use the info() during init.
+            impl: ActivityInboundInterceptor = _ActivityInboundImpl()
+            for interceptor in reversed(list(self._config["interceptors"])):
+                impl = interceptor.intercept_activity(impl)
+            # Init
+            impl.init(_ActivityOutboundImpl(self, info))
+            # Exec
+            result = await impl.execute_activity(input)
+            # Convert result if not none. Since Python essentially only supports
+            # single result types (even if they are tuples), we will do the
+            # same.
+            if result is None:
+                completion.result.completed.SetInParent()
+            else:
+                result_payloads = await self._config["client"].data_converter.encode(
+                    [result]
+                )
+                # We have to convert from Temporal API payload to Core payload
+                completion.result.completed.result.metadata.update(
+                    result_payloads[0].metadata
+                )
+                completion.result.completed.result.data = result_payloads[0].data
+        except (BaseException, asyncio.CancelledError) as err:
+            try:
+                if isinstance(err, temporalio.activity.CompleteAsyncError):
```

Comment:
Notice I have this entire block in its own try just in case something happens with exception handling. Extra try blocks just make that more difficult by adding multiple exception handling blocks instead of just one (even though this shouldn't throw, the point still stands for all exceptions herein...one block = clarity).